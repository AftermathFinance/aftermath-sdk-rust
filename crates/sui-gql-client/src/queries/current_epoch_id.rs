use af_sui_types::Version;
use cynic::{GraphQlResponse, QueryFragment};

use super::Error;
use crate::{missing_data, schema, GraphQlClient, GraphQlResponseExt as _};

pub async fn query<C: GraphQlClient>(client: &C) -> Result<u64, Error<C::Error>> {
    let curr: GraphQlResponse<Query> = client.query(()).await.map_err(Error::Client)?;
    let curr_epoch_id = curr
        .try_into_data()?
        .ok_or_else(|| missing_data!("No data"))?
        .epoch
        .ok_or_else(|| missing_data!("epoch"))?
        .epoch_id;
    Ok(curr_epoch_id)
}

#[cfg(test)]
#[allow(clippy::unwrap_used)]
#[test]
fn init_gql_output() {
    use cynic::QueryBuilder as _;
    let operation = Query::build(());
    insta::assert_snapshot!(operation.query);
}

#[derive(QueryFragment, Clone, Debug)]
#[cynic(graphql_type = "Query")]
struct Query {
    epoch: Option<Epoch>,
}

#[derive(QueryFragment, Clone, Debug)]
struct Epoch {
    epoch_id: Version,
}
