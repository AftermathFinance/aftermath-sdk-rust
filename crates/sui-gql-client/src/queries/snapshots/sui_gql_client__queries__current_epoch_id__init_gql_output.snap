---
source: crates/sui-gql-client/src/queries/current_epoch_id.rs
expression: operation.query
---
query Query {
  epoch {
    epochId
  }
}
